core/domain/story_services.py

```diff
@@ -101,16 +101,28 @@ def apply_change_list(story_id, change_list):
             story.
 
     Returns:
-        Story. The resulting story domain object.
+        Story, list(str), list(str). The resulting story domain object, the
+            exploration IDs removed from story and the exploration IDs added to
+            the story.
     """
     story = story_fetchers.get_story_by_id(story_id)
+    exp_ids_removed_from_story = []
+    exp_ids_added_to_story = []
     try:
         for change in change_list:
             if not isinstance(change, story_domain.StoryChange):
                 raise Exception('Expected change to be of type StoryChange')
             if change.cmd == story_domain.CMD_ADD_STORY_NODE:
                 story.add_node(change.node_id, change.title)
             elif change.cmd == story_domain.CMD_DELETE_STORY_NODE:
+                for node in story.story_contents.nodes:
+                    if node.id == change.node_id:
+                        if node.exploration_id in exp_ids_added_to_story:
+                            exp_ids_added_to_story = [
+                                exp_id for exp_id in exp_ids_added_to_story
+                                if exp_id != node.exploration_id]
+                        exp_ids_removed_from_story.append(node.exploration_id)
```

Comment:
Hmm, isn't that fine? Since, that exploration is removed from the story, with the updates till that point. Should I rename the array `exp_ids_removed_from_story` since it doesn't represent the exp IDs removed from the story as present in the datastore, but more like the temporary exp IDs changes added to and removed from the intermediate story object....

Reply:
Well, that's the thing -- I think that array should actually end up representing the exp IDs removed from the story as present in the datastore, right? You're right that it currently seems to be representing the temporary thing you're talking about, but I don't think that's what we actually want...

Reply:
Ok, done. Modified to represent the actual removal. Here is the output (removed from story and added to story) for the 4 test cases:
![image](https://user-images.githubusercontent.com/22347970/75610005-7f92ac80-5b33-11ea-9cc7-900aa3e47e3b.png)